WebApp/WebContent/data_file/upload.xhtml

```diff
@@ -31,55 +31,116 @@
-					rowStyleClass="#{file.messageClass}"
+					rowStyleClass="#{uploadedFile.messageClass}"
 					rowIndexVar="rowIndex"
 					>
 					    <p:column headerText="Filename">
-					        <h:outputText value="#{file.name}" />
+					        <h:outputText value="#{uploadedFile.name}" />
 					    </p:column>
 					    <p:column headerText="Start date">
 					        <h:outputText escape="false"
-							value="#{file.startDate}" >
+							value="#{uploadedFile.startDate}" >
 							    <f:convertDateTime pattern="#{fileUpload.longDateFormat}" />
 							</h:outputText>
-							<div class="tablespinner #{file.startDate == null and file.messageClass == '' ? 'loading' : ''}"></div>
+							<div class="tablespinner #{uploadedFile.startDate == null and uploadedFile.messageClass == '' ? 'loading' : ''}"></div>
 					    </p:column>
 
 					    <p:column headerText="End date">
 					        <h:outputText escape="false"
-							value="#{file.endDate}" >
+							value="#{uploadedFile.endDate}" >
 							    <f:convertDateTime pattern="#{fileUpload.longDateFormat}" />
 							</h:outputText>
-							<div class="tablespinner #{file.endDate == null and file.messageClass == '' ? 'loading' : ''}"></div>
+							<div class="tablespinner #{uploadedFile.endDate == null and uploadedFile.messageClass == '' ? 'loading' : ''}"></div>
 					    </p:column>
 
 					    <p:column headerText="No. of records">
-					        <h:outputText value="#{file.rowCount}" />
+					        <h:outputText value="#{uploadedFile.rowCount}" />
 					    </p:column>
 					    <p:column headerText="Store file to database">
 							<p:selectBooleanCheckbox
 							id="fileStoreCheckbox"
-							styleClass="#{file.endDate == null or file.messageClass != '' ? 'hidden' : 'ready'}"
-							value="#{file.store}" />
+							styleClass="#{uploadedFile.endDate == null or uploadedFile.messageClass != '' ? 'hidden' : 'ready'}"
+							value="#{uploadedFile.store}" />
 							<p:commandButton
-							id="messageButton"
-							icon="fa-info-circle"
-							styleClass="messageButton #{file.messageClass == '' ? 'hidden' : ''}"
+							icon="ui-icon-notice"
+							styleClass="messageButton #{uploadedFile.messageClass == '' ? 'hidden' : ''}"
 							onclick="renderMessages($(this).data('messages'))"
 							title="Errors"
 							value="Errors"
 							>
-								<f:passThroughAttribute name="data-messages" value="#{file.messages}" />
+								<f:passThroughAttribute name="data-messages" value="#{uploadedFile.messages}" />
+							</p:commandButton>
+							<p:commandButton
+							icon="ui-icon-plus"
+							styleClass="missingRunTypeButton #{uploadedFile.missingRunTypeClass}"
+							onclick="PF('runTypesDialog_#{rowIndex}').show()"
+							title="Manage Missing Run Types"
+							value="Manage Missing Run Types"
+							>
 							</p:commandButton>
+						    <p:dialog widgetVar="runTypesDialog_#{rowIndex}">
+								<div class="runTypeFile">
+									<p:fieldset legend="Update undefined Run Types">
```

Comment:
I think the dialog needs a short sentence describing what the user's doing. Something like:

_This file contained unrecognised Run Types. Please assign them using the menu options._

There's a CSS class `instructions` that's used for these messages.

Then the `fieldset` title can change to "Unrecognised Run Types"